src/baldr/graphtile.cc

```diff
@@ -38,6 +37,19 @@ struct dir_facet : public std::numpunct<char> {
 const std::locale dir_locale(std::locale("C"), new dir_facet());
 const AABB2<PointLL> world_box(PointLL(-180, -90), PointLL(180, 90));
 constexpr float COMPRESSION_HINT = 3.5f;
+
+std::string MakeSingleTileUrl(const std::string& tile_url, const valhalla::baldr::GraphId& graphid) {
+  auto id_pos = tile_url.find(valhalla::baldr::GraphTile::kTilePathPattern);
+  return tile_url.substr(0, id_pos) + valhalla::baldr::GraphTile::FileSuffix(graphid.Tile_Base()) +
+         tile_url.substr(id_pos + std::strlen(valhalla::baldr::GraphTile::kTilePathPattern));
+}
+
+std::string GenerateTmpSuffix() {
+  // TODO: use /dev/random or similar on windows
```

Comment:
we can make this more secure/intellegent. i still want to do the right thing here before merging this PR